plugin-process-to-cgroup-bridge/src/tests.rs

```diff
@@ -0,0 +1,437 @@
+        };
+
+        let check_output = move |measurements: &MeasurementBuffer| {
+            assert_measurement_counts(measurements, expected_counts.clone());
+        };
+
+        let runtime_expectations = RuntimeExpectations::new().test_transform(
+            TransformName::from_str("process-to-cgroup-bridge", "transform"),
+            make_input,
+            check_output,
+        );
+
+        let agent = agent::Builder::new(plugins)
+            .with_expectations(runtime_expectations)
+            .build_and_start()
+            .unwrap();
+
+        agent.wait_for_shutdown(Duration::from_secs(2)).unwrap();
+
+        Ok(())
+    }
+
+    #[test]
+    fn test_default_setup() -> anyhow::Result<()> {
+        let config = Config {
+            processes_metrics: vec!["metric_a".to_string(), "metric_b".to_string()],
+            ..Default::default()
+        };
+
+        let expected = ExpectedCounts {
+            t_initial: 6,
+            t2_initial: 1,
+            t_shared: 2,
+            t_single: 1,
+            t2_shared: 1,
+            t2_single: 0,
+        };
+
+        run_test_with_config(config, expected)
+    }
+
+    #[test]
+    fn test_only_metric_a() -> anyhow::Result<()> {
+        let config = Config {
+            processes_metrics: vec!["metric_a".to_string()],
+            ..Default::default()
+        };
+
+        let expected = ExpectedCounts {
+            t_initial: 6,
+            t2_initial: 1,
+            t_shared: 1,
+            t_single: 1,
+            t2_shared: 1,
+            t2_single: 0,
+        };
+
+        run_test_with_config(config, expected)
+    }
+
+    #[test]
+    fn test_merge_config_disable() -> anyhow::Result<()> {
+        let config = Config {
+            processes_metrics: vec!["metric_a".to_string(), "metric_b".to_string()],
+            merge_similar_cgroups: false,
+            ..Default::default()
+        };
+
+        let expected = ExpectedCounts {
+            t_initial: 6,
+            t2_initial: 1,
+            t_shared: 4,
+            t_single: 1,
+            t2_shared: 1,
+            t2_single: 0,
+        };
+
+        run_test_with_config(config, expected)
+    }
+
+    #[test]
+    fn test_keep_config_disable() -> anyhow::Result<()> {
+        let config = Config {
+            processes_metrics: vec!["metric_a".to_string(), "metric_b".to_string()],
+            keep_processed_measurements: false,
+            ..Default::default()
+        };
+
+        let expected = ExpectedCounts {
+            t_initial: 1,
+            t2_initial: 0,
+            t_shared: 2,
+            t_single: 1,
+            t2_shared: 1,
+            t2_single: 0,
+        };
+
+        run_test_with_config(config, expected)
+    }
+
+    #[test]
+    fn test_process_id_not_found_in_procfs() -> anyhow::Result<()> {
+        let base_path = prepare_procfs_mock()?;
+        set_proc_path_for_test(base_path);
```

Comment:
Tests run in parallel, and `set_proc_path_for_test` will only initialize the global variable once (the first test wins). Therefore, the `base_path` is only correct for the test that wins the race, which is non deterministic.
If you intend the first test to win, `prepare_procfs_mock` should be called only once, on initialization. I would replace the two lines with something like
```rs
let base_path = prepare_procfs_for_test()
```

Reply:
Removed the global variable that was shared by all tests. Instead relying on configuration only available for tests.
We had multiple issues by using a global variable as tests are run in //. We were not able to make multiple mocks for multiple tests. And also it was complicated to have TmpDir deleted as the Dir was shared by all.